### PR TITLE
Burger Variation Sprites

### PR DESCRIPTION
**Pull request by CoolKakapo**
## About the PR 
Adds special sprites for the synthbuttburger, synthbrainburger, synthheartburger, cyberbuttburger, cyberheartburger, Spontaneous Intelligence Creation Core burger, flockbrainburger and the flockheartburger. Also adds a special sprite for the normal synthburger. The burgers are considered new foods by the code but the status effects and special properties stay the same for each variation.

## Why's this needed? 
It will be more fun to use different ingredients for burgers, and makes burgers feel more special.

## Changelog 

```changelog
(u)The_Kakapo
(*)Adds special sprites for different ingredients used in brain/heart/buttburgers
(+)You can make flock heart/brainburgers, synthbutt/brain/heartburgers and cyberbutt/heartburgers and Spontaneous Intelligence Creation Core burgers.
```

### CODE REVIEW

**Comment by TobleroneSwordfish** [comment]
Sorry, we're currently in a feature freeze, please re-submit this next month.

**Comment by BatElite** [comment]
Freeze over, Kakapo wanted this reopened

**Review by TobleroneSwordfish — CHANGES_REQUESTED** [review]
Sprites look great to me, but I'll let a sprite dev give their opinion before merging.
Also the different burger types should probably be subtypes of their respective burger, ie `/obj/item/reagent_containers/food/snacks/burger/buttburger/synth` instead of `/obj/item/reagent_containers/food/snacks/burger/synthbuttburger` to cut down the copy-paste.

**Comment by CoolKakapo** [comment]
The burgers are now a subtype of their normal burgers that were already in the game.

File: code/modules/food_and_drink/sandwiches.dm [hunk 1]
```diff
@@ -243,14 +266,71 @@
...
+/obj/item/reagent_containers/food/snacks/burger/heartburger/synth
+	name = "heartburger"
+	desc = "A hearty meal, made with Love. This one seems to contain a green synthetic heart."
+	icon_state = "synthheartburger"
+	food_effects = list("food_sweaty_big", "food_hp_up_big")
+	meal_time_flags = MEAL_TIME_LUNCH | MEAL_TIME_DINNER
+
+	New()
+		..()
+		reagents.add_reagent("love", 15)
```
Comment (TobleroneSwordfish) [inline code/modules/food_and_drink/sandwiches.dm:278]: You can remove anything from these child types that's the same on the parent, that's the reason for making them child types in the first place!

**Review by TobleroneSwordfish — CHANGES_REQUESTED** [review]

File: code/modules/food_and_drink/sandwiches.dm (hunk 1 above)
Comment (CoolKakapo) [reply code/modules/food_and_drink/sandwiches.dm]: did it, i also fixed a bug where making a flockheart/brainburger would result in a regular flockburger, which now makes the correct burger

File: code/modules/food_and_drink/sandwiches.dm [hunk 2]
```diff
@@ -231,10 +231,19 @@
 		if(prob(10))
 			name = pick("cleveland steamed ham","very sloppy joe","buttconator","bootyburg","quarter-mooner","ass whooper","hambuttger","big crack")
 
+/obj/item/reagent_containers/food/snacks/burger/buttburger/synth
+	name = "buttburger"
+	desc = "This burger's all buns. It seems to be made out of a green synthetic butt."
+	icon_state = "assburger"
```
Comment (TobleroneSwordfish) [inline code/modules/food_and_drink/sandwiches.dm:237]: shouldn't this be a different icon state?

File: code/modules/food_and_drink/sandwiches.dm [hunk 3]
```diff
@@ -343,7 +382,7 @@
 /obj/item/reagent_containers/food/snacks/burger/synthburger
 	name = "synthburger"
 	desc = "A thoroughly artificial snack."
-	icon_state = "hburger"
+	icon_state = "Synthburger"
```
Comment (TobleroneSwordfish) [inline code/modules/food_and_drink/sandwiches.dm:385]: would prefer no uppercase icon state names

**Review by TobleroneSwordfish — CHANGES_REQUESTED** [review]

**Review by TobleroneSwordfish — APPROVED** [review]

**Pull request merged**